### DISCUSSION TITLE
Supabase storage - delete all files in the folder

### DISCUSSION THREAD

**Discussion by Mikilll94** [supabase/supabase#15648] [Feature Requests]
Currently there is no easy way to remove all files from a folder or to remove a folder.

We should add a possibility to write code like this:

```js
// remove all files from the folder
const { data, error } = await supabase
  .storage
  .from('avatars')
  .remove(['folder/*']);
```

```js
// remove folder
const { data, error } = await supabase
  .storage
  .from('avatars')
  .remove(['folder'])
```

**Comment by pierroo** [comment]
Did you find any solution to this @Mikilll94 or is there really none?
I am trying to have a function triggered by a database delete, to also remove all associated storage files (from specific folder)
